### ISSUE TITLE
Pass request object to limiter callbacks

### ISSUE THREAD

**Issue by dlindenkreuz** [laurentS/slowapi#41]
The request object cannot be accessed easily in some of the limiter callbacks (`limit_value`, `exempt_when`). Support for this would be really useful. This already seems to work with `key_func` which gets passed the `request` if the callback declares a `request` parameter.

This would also remove the need for the workaround from https://github.com/laurentS/slowapi/issues/13.

**Comment by Kavan72** [comment]
@laurentS do you have any plan to add this is near future ?

**Comment by fredi-python** [comment]
Are there any news to that?

**Comment by iursevla** [comment]
Yeah, this is mandatory if we want to get some configuration from the request
